From 4057f6ee1f4d3824641d50bc6393be62b5e7a0c0 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Wed, 5 Sep 2012 13:26:50 +0800
Subject: [PATCH 2013/2463] ENGR00223056 Fix HDMI build warning

Fix HDMI build warning.
drivers/video/mxc_hdmi.c: In function 'mxc_hdmi_set_mode':
drivers/video/mxc_hdmi.c:1659: warning: assignment discards
 qualifiers from pointer target type
drivers/video/mxc_hdmi.c: At top level:
driver/video/mxc_hdmi.c:1398: warning: 'mxc_hdmi_enable_pins'
 defined but not used

Remove unused function mxc_hdmi_enable_pins() and mxc_hdmi_disable_pins()
from code. Fix defined but unused function build warning.
Added pointer conversion from const poniter to non-const pointer.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/video/mxc_hdmi.c |   26 +-------------------------
 1 files changed, 1 insertions(+), 25 deletions(-)

diff --git a/drivers/video/mxc_hdmi.c b/drivers/video/mxc_hdmi.c
index e9c0547..a45c347 100644
--- a/drivers/video/mxc_hdmi.c
+++ b/drivers/video/mxc_hdmi.c
@@ -1394,28 +1394,6 @@ static int mxc_hdmi_read_edid(struct mxc_hdmi *hdmi)
 	return HDMI_EDID_SUCCESS;
 }
 
-static void mxc_hdmi_enable_pins(struct mxc_hdmi *hdmi)
-{
-	struct fsl_mxc_hdmi_platform_data *plat = hdmi->pdev->dev.platform_data;
-
-	dev_dbg(&hdmi->pdev->dev, "%s\n", __func__);
-
-	/* Enable pins to HDMI */
-	if (plat->enable_pins)
-		plat->enable_pins();
-}
-
-static void mxc_hdmi_disable_pins(struct mxc_hdmi *hdmi)
-{
-	struct fsl_mxc_hdmi_platform_data *plat = hdmi->pdev->dev.platform_data;
-
-	dev_dbg(&hdmi->pdev->dev, "%s\n", __func__);
-
-	/* Disable pins to HDMI */
-	if (plat->disable_pins)
-		plat->disable_pins();
-}
-
 static void mxc_hdmi_phy_disable(struct mxc_hdmi *hdmi)
 {
 	dev_dbg(&hdmi->pdev->dev, "%s\n", __func__);
@@ -1656,7 +1634,7 @@ static void mxc_hdmi_set_mode(struct mxc_hdmi *hdmi)
 		dev_dbg(&hdmi->pdev->dev,
 				"%s: Video mode same as previous\n", __func__);
 		/* update fbi mode in case modelist is updated */
-		hdmi->fbi->mode = mode;
+		hdmi->fbi->mode = (struct fb_videomode *)mode;
 		mxc_hdmi_phy_init(hdmi);
 	} else {
 		dev_dbg(&hdmi->pdev->dev, "%s: New video mode\n", __func__);
@@ -2297,8 +2275,6 @@ static void mxc_hdmi_disp_deinit(struct mxc_dispdrv_handle *disp)
 
 	fb_unregister_client(&hdmi->nb);
 
-	mxc_hdmi_disable_pins(hdmi);
-
 	clk_disable(hdmi->hdmi_isfr_clk);
 	clk_put(hdmi->hdmi_isfr_clk);
 	clk_disable(hdmi->hdmi_iahb_clk);
-- 
1.7.7.4

